src/ScottPlot5/ScottPlot/Control/Backend.cs

```diff
@@ -0,0 +1,187 @@
+﻿using ScottPlot.Control.EventArgs;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+
+namespace ScottPlot.Control
+{
+    public class Backend
+    {
+        private MouseDownEventArgs? CurrentMouse1Down = null;
+        private MouseDownEventArgs? CurrentMouse2Down = null;
+        private MouseDownEventArgs? CurrentMouse3Down = null;
+
+        private readonly HashSet<Key> CurrentlyPressedKeys = new();
+        private Pixel? LastMousePosition = null;
+
+        private readonly Plot plot;
+        private readonly object EventSender;
+        private readonly Action requestRender;
+
+        public delegate void MouseDownHandler(object sender, MouseDownEventArgs e);
+        public delegate void MouseMoveHandler(object sender, MouseMoveEventArgs e);
+        public delegate void MouseUpHandler(object sender, MouseUpEventArgs e);
+        public delegate void MouseDragHandler(object sender, MouseDragEventArgs e);
+        public delegate void MouseDragEndHandler(object sender, MouseDragEventArgs e);
+        public delegate void DoubleClickHandler(object sender, MouseDownEventArgs e);
+        public delegate void MouseWheelHandler(object sender, MouseWheelEventArgs e);
+        public delegate void KeyDownHandler(object sender, KeyDownEventArgs e);
+        public delegate void KeyUpHandler(object sender, KeyUpEventArgs e);
+
+        public event MouseDownHandler MouseDown = delegate { };
```

Comment:
To differentiate this big list of ScottPlot events from the typical events controls may have, do you think we should name them things like `SPMouseDown`? Or group them all in a class? A class would actually make it easier to duplicate all these across controls 🤔

Reply:
My bad - ignore that comment. I got confused and thought I was reading user control code (not backend code) 🤪